ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -700,6 +771,12 @@ VALUE Message_decode(VALUE klass, VALUE data) {
 
   stackenv_uninit(&se);
 
+  if (desc->has_udts_transitively) {
```

Comment:
Any reason we have to wait until the end and do a totally separate pass? Couldn't this happen in the end handlers for the specific subtree that needs conversion?